### ISSUE TITLE
ReporteRs under knitr - hangs the whole process

### ISSUE THREAD

**Issue by alliquot** [davidgohel/ReporteRs#60]
Could you try to put some code that generates simplest DOCX document inside a code chunk in RMD (RMarkdown) file and run it from under RStudio?
 
It could be just:
doc = docx()
doc = addParagraph(doc, "Hey")
writeDoc( doc, file = "output.docx")

(I don't know how to put leading and trailing rmarkdown chunk tag: ```{ r } as it is interpreted)
 
The documents generates correctly, but it hanges RStudio permanently or crashes.
I'm using knitr in version 2 (based on pandoc).

EDIT: perhaps this causes the problem?
http://stackoverflow.com/questions/13295130/knitr-hanging-on-system-call

**Comment by davidgohel** [comment]
I tried the following code but I am not able to reproduce your issue (I didn't find knitr V2, I am with the version 1.8)


    ```{r}
    library( ReporteRs )
    doc = docx()
    doc = addParagraph(doc, "Hey")
    writeDoc( doc, file = "D:/output.docx")
    ```

**Comment by alliquot** [comment]
I mean the version that uses pandoc and is implemented in current RStudio.

Steps to reproduce this:
1. Add RMarkdown file
2. write the code
3. click: "Knit HTML"

It calls function "render" which hangs right after the document is generated. When I run the code outside RStudio, calling the function render() via rscript.exe, the issue remains.

I've tried this on 3 different machines and even virtual machine - with the same result. I'm near the finish of my 8 month task and stuck with this issue at the very end...

**Comment by davidgohel** [comment]
sorry for you...

I have just tried with "Knit HTML" and with render() and had no problem.

**Comment by davidgohel** [comment]
OK, I just read your edit. 

If you find something interesting, could you tell me?

**Comment by alliquot** [comment]
So, after a long investigation... My code stopped hanging when I removed "results='asis'" from the chunk with ReporteRs code. The code calls a lot of functions transforming data and generating tables. I cannot paste it since it's a professional secret, and those are too complex to make a "mockups". Then I print the status in HTML (the reason for "results='asis'"). So, when I put it all together, the code hangs  and CPU usage reaches 100%. When I moved the HTML code with "results=asis" to another chunk, it started working properly - now it smoothly generates docx in one chunk and prints the result in other. Also, when I remove (comment) any calls of RepporteRs - it passes correctly as well. Don't have any idea what causes this, but I'm happy I could solve this.

In simple, exemplary code as above the problem doesn't exist. I don't know what causes the problem
